sub-packages/bionemo-llm/src/bionemo/llm/data/datamodule.py

```diff
@@ -0,0 +1,69 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: LicenseRef-Apache2
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from typing import Any, Dict
+
+import pytorch_lightning as pl
+from nemo.utils import logging
+
+
+class MegatronDatamodule(pl.LightningDataModule):
+    """A mixin that adds a `state_dict` and `load_state_dict` method for datamodule training resumption in NeMo."""
+
+    def __post_init__(self):
+        """Set init_global_step to 0 for datamodule resumption."""
+        self.init_global_step = 0
```

Comment:
@sichu2023 this is a cool idea, but I don't think `__post_init__` gets called outside of in `@datamodule` classes? Double check that.

Reply:
Ya. This is only for `dataclass`.